### PR TITLE
[WIP]feat:  add hadoop3.3, spark3.2, hive3.1, scala2.12 profile and unify some maven plugin 

### LINKED ISSUE #1964: [Feature] code import orders format not work bug
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-linkis/issues) and found no similar feature requirement.

### Problem Description

@see [spotless-maven-plugin](https://github.com/diffplug/spotless/issues/731)   the not worked as the plugin config
```
<importOrder>
    <order>org.apache.linkis,org.apache,org.springframework,javax,java,scala,\#</order>
</importOrder>
```

and the order string should be 
```
 <importOrder>
      <order>org.apache.linkis,org.apache,org.springframework,javax,java,scala,,\#</order>
  </importOrder>
```

### Description

the order in the java file is not current in the linkis-metadata module's HiveUtils https://github.com/apache/incubator-linkis/blob/master/linkis-public-enhancements/linkis-datasource/linkis-metadata/src/main/java/org/apache/linkis/metadata/util/HiveUtils.java the import order is

```java
import org.apache.commons.lang.StringUtils;
import org.apache.hadoop.conf.Configuration;
import org.apache.hadoop.fs.Path;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.File;

import sun.misc.BASE64Decoder;
```

### Use case

_No response_

### solutions

_No response_

### Anything else

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by jackxu2011**
### What is the purpose of the change
add hadoop-3.3, spark-3.2 , hive-3.1 and scala-2.12 profile and unify some plugin version

### Brief change log
1. add hadoop-3.3, spark-3.2 , hive-3.1 and scala-2.12 profile
 2. unify hadoop, spark, hive version to parent pom
 3. unify mysql version to 8.0.16
 4. unify maven-deploy-plugin to 3.0.0-M1
 5. unify maven-assembly-plugin to 3.2.0
 6. unify maven.compiler-plugin to 3.8.1
 7. upgrade scala-maven-plugin to 4.3.2

### Verifying this change
(Please pick either of the following options)  
This change is a trivial rework / code cleanup without any test coverage.  

### Does this pull request potentially affect one of the following parts:
- Dependencies (does it add or upgrade a dependency): ( no)
- Anything that affects deployment: (no)
- The MGS(Microservice Governance Services), i.e., Spring Cloud Gateway, OpenFeign, Eureka.: (no)

### Documentation
- Does this pull request introduce a new feature? ( no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by jackxu2011** [comment]
1. 以后修改引擎版本，hadoop, spark, hive的引擎版本就都在parent的pom文件中
2. 可以用命令 mvn clean install -Pspark-3.2 -Phive-3.1 -Phadoop-3.3 打包， 但是在打包前要改一个spark3.2的代码兼容问题，在profile中有说明

File: linkis-engineconn-plugins/engineconn-plugins/hive/pom.xml [hunk 1]
```diff
@@ -208,89 +204,6 @@
         <dependency>
             <groupId>org.apache.hadoop</groupId>
             <artifactId>hadoop-client</artifactId>
-            <version>${hadoop.version}</version>
-            <exclusions>
```
Comment (legendtkl) [inline linkis-engineconn-plugins/engineconn-plugins/hive/pom.xml:212]: why remove the exclusions

File: linkis-commons/linkis-common/src/main/scala/org/apache/linkis/common/conf/CommonVars.scala [hunk 2]
```diff
@@ -17,14 +17,18 @@
...
+import java.util.Properties
 import scala.collection.JavaConversions._
 
 
 case class CommonVars[T](key: String, defaultValue: T, value: T, description: String = null) {
   val getValue: T = BDPConfiguration.getOption(this).getOrElse(defaultValue)
   def getValue(properties: java.util.Map[String, String]): T = {
-    if(properties == null || !properties.containsKey(key) || properties.get(key) == null) getValue
-    else BDPConfiguration.formatValue(defaultValue, Option(properties.get(key))).get
+    if (!(properties == null || !properties.containsKey(key)) && !(properties.get(key) == null)) {
```
Comment (legendtkl) [inline linkis-commons/linkis-common/src/main/scala/org/apache/linkis/common/conf/CommonVars.scala:27]: the refactor logic looks more confused than before, why do this refactor?

**Review by legendtkl — COMMENTED** [review]
Overall LGTM with minor comments.

File: linkis-engineconn-plugins/engineconn-plugins/hive/pom.xml (hunk 1 above)
Comment (jackxu2011) [reply linkis-engineconn-plugins/engineconn-plugins/hive/pom.xml:212]: 统一在parent中进行排除，不用在每一个模块中进行排除

File: linkis-commons/linkis-common/src/main/scala/org/apache/linkis/common/conf/CommonVars.scala (hunk 2 above)
Comment (jackxu2011) [reply linkis-commons/linkis-common/src/main/scala/org/apache/linkis/common/conf/CommonVars.scala:27]: > the refactor logic looks more confused than before, why do this refactor?

减少代码复杂度， 其实可读性应该没有降低多少

Comment (legendtkl) [reply linkis-commons/linkis-common/src/main/scala/org/apache/linkis/common/conf/CommonVars.scala:27]: I prefer the code before changed. What do you think ? @peacewong  @casionone

**Comment by legendtkl** [comment]
LGTM

**Comment by casionone** [comment]
As discussed
Major changes require basic engine regression testing. We need to organize a regression test baseline. After verification, there is no problem, and then merge

--------
如我们所讨论的
大的变动，需要进行基本引擎回归测试，我们正在整理一个回归测试基线 验证没问题后， 然后再合并

**Pull request closed without merging**